src/core/lib/pick-layers.js

```diff
@@ -348,42 +348,53 @@ function getClosestFromPickingBuffer(gl, {
   deviceRect
 }) {
   assert(pickedColors);
-  let closestResultToCenter = NO_PICKED_OBJECT;
 
   // Traverse all pixels in picking results and find the one closest to the supplied
   // [deviceX, deviceY]
+  const {x, y, width, height} = deviceRect;
   let minSquareDistanceToCenter = deviceRadius * deviceRadius;
+  let closestPixelIndex = -1;
   let i = 0;
 
-  for (let row = 0; row < deviceRect.height; row++) {
-    for (let col = 0; col < deviceRect.width; col++) {
-      // Decode picked layer from color
-      const pickedLayerIndex = pickedColors[i + 3] - 1;
-
-      if (pickedLayerIndex >= 0) {
-        const dx = col + deviceRect.x - deviceX;
-        const dy = row + deviceRect.y - deviceY;
-        const d2 = dx * dx + dy * dy;
-
-        if (d2 <= minSquareDistanceToCenter) {
-          minSquareDistanceToCenter = d2;
-
-          // Decode picked object index from color
-          const pickedColor = pickedColors.slice(i, i + 4);
-          const pickedLayer = layers[pickedLayerIndex];
-          if (pickedLayer) {
-            const pickedObjectIndex = pickedLayer.decodePickingColor(pickedColor);
-            closestResultToCenter = {pickedColor, pickedLayer, pickedObjectIndex};
-          } else {
-            log.error(0, 'Picked non-existent layer. Is picking buffer corrupt?');
+  for (let row = 0; row < height; row++) {
+    const dy = row + y - deviceY;
+    const dy2 = dy * dy;
+
+    if (dy2 > minSquareDistanceToCenter) {
+      // skip this row
```

Comment:
`deviceRect` is calculated to so that picking circle fits in it exactly, I wonder how many times we skip an entire row, may be just once? if so , should we just adjust calculating `deviceRect`,   reduce width and height by 1.

Reply:
`minSquareDistanceToCenter` reflects the distance between the last best hit and the pointer position. So best case, if there is an object at `[deviceX, deviceY]`, the second half of `deviceRect` is skipped.